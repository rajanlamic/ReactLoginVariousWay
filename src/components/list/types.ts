export interface ListData {
    email: string,
    first_name: string,
    last_name: string
    title: string,
    user_name: string
    _id: string,
}

export interface ListState {
    isPening?: boolean,
    isFulfilled?: boolean,
    isRejected?: boolean,
    myList?: ListData[]
}
