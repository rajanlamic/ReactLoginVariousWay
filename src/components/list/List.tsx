import React from 'react'
import { ListData } from './types'

export interface ListProps {
    list: ListData[]
}

export const Lists: React.FC<ListProps> = ({ list = [] }) => {
    return <ul>
        {
            list.map(item => {
                return <li key={item.first_name}>{item.first_name}</li>
            })
        }
    </ul>
}

// export default Lists
