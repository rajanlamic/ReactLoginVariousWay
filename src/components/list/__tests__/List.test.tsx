import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme';
import { Lists, ListProps } from '../List'
import { ListData } from '../types'

describe("List componnet", () => {
    let wrapper: any, props: any;

    beforeEach(() => {
        props = {
            list: [{
                email: "rajanlamic@gmail.com",
                first_name: "rajan",
                last_name: "lamichhane",
                title: "mr",
                user_name: "rajanlkamic",
                _id: "1",
            },
            {
                email: "rajanlamic1@gmail.com",
                first_name: "nextguy",
                last_name: "lamichhane1",
                title: "mr1",
                user_name: "rajanlkamic1",
                _id: "11",
            }
            ]
        }
        wrapper = shallow(<Lists {...props} />)
    })

    afterEach(() => {
        props = null;
        wrapper = null;
    })

    it("should render Lists properly with snapshot", () => {
        expect(wrapper).toMatchSnapshot()
    })

    it("should render ul properly", () => {
        expect(wrapper.find("ul").length).toEqual(1)
    })

    it("should render 2 li element", () => {
        expect(wrapper.find("li").length).toEqual(2)
    })

    it("should pass right parameters in the list", () => {
        expect(wrapper.find("li").first().props().children).toBe("rajan")
        expect(wrapper.find("li").at(1).props().children).toBe("nextguy")
    })

})
