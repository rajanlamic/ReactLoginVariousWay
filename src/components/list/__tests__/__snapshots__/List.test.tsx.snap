// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List componnet should render Lists properly with snapshot 1`] = `ShallowWrapper {}`;
